src/components/molecules/ScheduleEvent/ScheduleEvent.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {
+  MouseEventHandler,
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import classNames from "classnames";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faBookmark as solidBookmark } from "@fortawesome/free-solid-svg-icons";
+import { faBookmark as regularBookmark } from "@fortawesome/free-regular-svg-icons";
+
+import { PersonalizedVenueEvent } from "types/venues";
+
+import { WithVenueId } from "utils/id";
+import { isTruthy } from "utils/types";
+import { isEventLive } from "utils/event";
+
+import { ONE_HOUR_IN_MINUTES } from "utils/time";
+
+import { saveEventToProfile } from "api/profile";
+import { useUser } from "hooks/useUser";
+
+import { calcStartPosition, HOUR_WIDTH } from "../Schedule/Schedule.utils";
+
+import "./ScheduleEvent.scss";
+
+export interface ScheduleEventProps {
+  event: WithVenueId<PersonalizedVenueEvent>;
+  scheduleStartHour: number;
+  isUsers?: boolean;
+}
+
+export const ScheduleEvent: React.FC<ScheduleEventProps> = ({
+  event,
+  scheduleStartHour,
+  isUsers,
+}) => {
+  const [isBookmarked, setBookmark] = useState(event.isSaved);
+
+  useEffect(() => {
+    setBookmark(event.isSaved);
+  }, [event.isSaved]);
```

Comment:
why do we need a local state? should we be using the remote state as a single source of truth?

Reply:
hmm for some reason decided that it won't rerender if I use just event.isSaved, but checked now and that worked. Thanks.
here is the commit: https://github.com/sparkletown/sparkle/pull/1302/commits/7bccbd028b48f3f9efb8beef30e9e86635010392